09_maps/main.go

```diff
@@ -111,18 +122,268 @@ func main() {
 
-	// TODO: Create a function that takes an array of maps as an argument and
-	// add as a new key-value pair to each map the key "bmi" with the value
-	// being the BMI of the corresponding person.
-	// Hint: Use a loop.
-	// Hint: Create a new map for each person.
-	// Hint: Create a new array of maps.
+	// add BMI to array of maps
+	fmt.Println(addBMI(people))
+
+	// test helper int function
+	fmt.Println(largestInt([]int{}))                      // 0
+	fmt.Println(largestInt([]int{1, 2, 3}))               // 3
+	fmt.Println(largestInt([]int{-1, -2, 0}))             // 0
+	fmt.Println(largestInt([]int{100, 0, 3000, 0, -100})) // 3000
+
+	// test helper string function
+	fmt.Println(largestString([]string{"", ""}))           // ""
+	fmt.Println(largestString([]string{"a", "bc", "def"})) // "def"
+
+	// Close main function, to be able to declare another function
+}
+
+// create empty map
+// create function
+// create counter
+// create loop
+// create break condition if counter is > than the key-value pairs the map has
+// operate on the input value of the keys and do math on it
+// add the result to a new map
+// add a new key value element and fill it depending on the value of the input height key
+// print new map
+
+var newMap map[string]int = map[string]int{}
+
+func createNewMap(mapInput map[string]int) {
+	var counter int = 0
+
+	for {
+		if counter > len(mapInput) {
+			break
+		}
+		newMap["age"] = mapInput["age"] + 2
+		newMap["height"] = mapInput["height"] * 2
+		newMap["weight"] = mapInput["weight"] * 2
+		if mapInput["height"] > 200 {
+			newMap["tall"] = 1
+		}
+		newMap["tall"] = 0
+		counter++
+	}
+	fmt.Println("createNewMap:", newMap)
+}
+
+// TODO: Create a function that takes a map as an argument and prints the
+// BMI of the person. The formula for BMI is:
+// BMI = weight / (height * height)
+// If the map does not have the key "height", print "Height not found".
+// If the map does not have the key "weight", print "Weight not found".
+// Otherwise, print "<name>'s BMI is <bmi>".
+// Hint: Use this function in later exercises.
+
+func calcBMI(mapInput map[string]int) float64 {
+	var value int
+	var exists bool
+	var bmi float64
+	var heightConvert int = mapInput["height"]
+	var heightConverted float64 = float64(heightConvert)
+	var weightConvert int = mapInput["weight"]
+	var weightConverted float64 = float64(weightConvert)
+
+	value, exists = mapInput["height"]
+	if !exists || value == 0 {
+		fmt.Println("Height not found")
+	}
+	value, exists = mapInput["weight"]
+	if !exists || value == 0 {
+		fmt.Println("Weight not found")
+	}
+	// debug
+	// fmt.Println("weight debug", mapInput["weight"])
+	// fmt.Println("height debug", mapInput["height"])
+	// debug
+	// fmt.Println(heightAdjust)
+	// fmt.Println(weightConverted)
+	bmi = (weightConverted / (heightConverted * heightConverted))
+	bmi = bmi * 10000
+	// debug
+	// fmt.Println(bmi)
+	// fmt.Println("BMI is", bmi)
+	return bmi
```

Comment:
Couple of comments
- Don't use `value`, but rather the variable containing the corresponding value (here `heightConvert`/`weightConvert`
- `heightConvert`/`weightConvert` are weird names, if you want to differentiate from `int` and `float`, using `heigthInt` and `heigthFloat` might be slightly better.
- Convert those to `float64` only _after_ you are sure they both exist

Reply:
Makes sense. I think I ran into some issues due to declaring variables before the assigned things existed.